src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbStreamsBySharedTable.java

```diff
@@ -159,68 +187,20 @@ private int retryLoadRecords(MtGetRecordsResult mtResult,
     private static int addRecords(List<? super MtRecord> mtRecords,
                                   int limit,
                                   List<? extends Record> records,
-                                  Function<Record, MtRecord> recordMapper,
-                                  Predicate<MtRecord> recordFilter) {
+                                  Predicate<Record> recordFilter,
+                                  Function<Record, MtRecord> recordMapper) {
         int consumedRecords = 0;
         for (Record record : records) {
-            MtRecord mtRecord = recordMapper.apply(record);
-            if (recordFilter.test(mtRecord)) {
+            if (recordFilter.test(record)) {
                 if (mtRecords.size() >= limit) {
                     break;
                 } else {
-                    mtRecords.add(mtRecord);
+                    mtRecords.add(recordMapper.apply(record));
                 }
             }
             consumedRecords++;
         }
         return consumedRecords;
     }
 
-    @Override
-    protected Predicate<MtRecord> getRecordFilter(StreamArn arn) {
-        Predicate<MtRecord> defaultPredicate = super.getRecordFilter(arn);
-        return mtRecord ->
-            mtDynamoDb.getMtContext().withContext(mtRecord.getContext(), this::isStreamEnabled, mtRecord.getTableName())
-                && defaultPredicate.test(mtRecord);
```

Comment:
One intentional side-effect of this change is that `isStreamEnabled` is no longer called as part of the record filter.

For calling with context, we already check whether the stream is enabled when we return the virtual ARN (in `describeTable`). We could check again at the beginning of `getRecords`, but tables are currently immutable, so it seems unnecessary.

For calling without context (KCL), we have observed that clients actually need to see all records. For example, KCL needs to see them to set checkpoints to the latest record in the physical stream. @sbabu-salesforce had also brought up the need to see all records for backup/restore. It could be argued that this somewhat breaks the abstraction, since the client now has to check on its end whether the virtual table for a given record has streaming enabled. On the other hand, we are already breaking the abstraction by allowing calling without context, so hopefully that's fine.

Reply:
Can we put some version of what you said above in the code comments?